app/models/card.rb

```diff
@@ -25,37 +19,39 @@ class Card < ActiveRecord::Base
   validate :not_equal_fields
   scope :for_review, -> { where("review_date < ?", Time.now) }
 
-  def check_translation(translation)
+  def check_translation(translation, time)
     l_distance = Text::Levenshtein.distance(translated_text, clear(translation))
     if l_distance < LEVENSHTEIN_DISTANCE_MAXIMUM
-      increase_review_level
       correct = true
+      get_next_interval(correct, time)
     else
-      decrease_review_level
       correct = false
+      get_next_interval(correct, time)
     end
     { correct: correct, levenshtein_distance: l_distance }
   end
 
-  def increase_review_level
-    self.error_counter = 0
-    if review_level < REVIEW_LEVELS
-      self.review_level += 1
-      new_date = Time.now + LEITNER_TIME[review_level]
-      self.review_date = new_date
+  def get_next_interval(correct, time)
+    sm = SuperMemoService.new
+    if correct
+      q = quality_from_timer(time)
+    else
+      q = 0
     end
+    next_step = sm.next_step(q, memo_ef, memo_interval)
+    self.memo_ef = next_step[:new_ef]
+    self.memo_interval = next_step[:new_i]
+    self.review_date = Time.now + memo_interval.seconds
```

Comment:
Do not use `Time.now` without zone. Use one of `Time.zone.now`, `Time.now.current`, `Time.now.in_time_zone`, `Time.now.utc`, `Time.now.getlocal`, `Time.now.iso8601`, `Time.now.jisx0301`, `Time.now.rfc3339`, `Time.now.to_i`, `Time.now.to_f` instead.